hw/ip/gpio/doc/theory_of_operation.md

```diff
@@ -73,14 +73,12 @@ the output driver is unable to switch the pin or during the delay imposed
 if the noise filter is enabled).
 
 #### GPIO Inputs as HW Straps
-The GPIO controller also provides an additional optional feature to sample the GPIO input values as hardware configuration straps
-These input values are sampled only at initial cold boot reset deassertion and is not sampled thereafter.
-Thus any change in the input configuration information after power on reset deassertion will not be captured.
-A strap_en signal driven by the pwrmgr is used for this sampling mechanism. The strap_en pulse is a single cycle pulse of the io div4 clock
-It is expected to toggle only once at cold boot.
-
-The snapshot value of the GPIO inputs is available in the [`HW_STRAPS_DATA_IN`](../data/gpio.hjson#hw_straps_data_in) register for firmware read out
-Note that all 32 bits of input data to the GPIO controller are sampled in this mode regardless of the gpio output enable status
+
+The GPIO controller includes an optional feature to sample GPIO input values as hardware configuration straps. After the reset (cold boot), when the strap_en signal is asserted, the GPIO detects the rising edge of the signal and triggers the sampling of GPIO input values one clock cycle later. These sampled values are stored in the HW_STRAPS_DATA_IN register.
+
+The strap_en signal is driven by the power manager (pwrmgr) and is expected to toggle only once after the reset process. Sampling occurs only during this period, and subsequent changes to the GPIO input configuration will not be captured.
```

Comment:
Is this chunk actually necessary? I think I'd be inclined to drop it, since that's very much a concern of stuff outside of the block.